src/cljs/org/broadinstitute/firecloud_ui/main.cljs

```diff
@@ -251,14 +254,48 @@
       [:div {}
        [comps/Button
         {:text "Sign In"
-         :href (str (config/api-url-root) "/login?callback="
-                    (js/encodeURIComponent
-                     (str js/window.location.protocol "//" js/window.location.hostname))
-                    (let [hash (nav/get-hash-value)]
-                      (if (clojure.string/blank? hash) "" (str "&path=" hash))))}]]
+         :onClick (fn [e]
+                    (.. js/window
+                        (open
+                         (str (config/api-url-root) "/login?callback="
+                              (js/encodeURIComponent (.. js/window -location -origin)))
+                         "Authentication"
+                         "menubar=no,toolbar=no,width=500,height=500")))}]]
       [:div {:style {:marginTop "1em"}} [RegisterLink]]
       [:div {:style {:maxWidth 600 :paddingTop "2em" :fontSize "small"}}
-        [Policy]]])})
+        [Policy]]])
+   :component-did-mount-post-message
+   (fn [{:keys [props locals]}]
+     (let [handle-message (fn [e] ((:on-login props) (.-data e)))]
+       (swap! locals assoc :message-listener handle-message)
+       (.addEventListener js/window "message" handle-message)))
+   :component-will-unmount-post-message
+   (fn [{:keys [locals]}]
+     (.removeEventListener js/window "message" (:message-listener @locals)))
+   :component-did-mount
+   (fn [{:keys [props locals]}]
+     (let [handle-message (fn [message] ((:on-login props) message))]
+       (aset js/window sign-in-handler-fn-name handle-message)))
+   :component-will-unmount
+   (fn [{:keys [locals]}]
+     (aset js/window sign-in-handler-fn-name nil))})
```

Comment:
(aset ... nil) or (js-delete ...)? It doesn't really matter, but delete is theoretically more correct?